set.mm

```diff
@@ -422891,24 +423014,30 @@ varying definitions (some start from 0, others start from 1), but
        details).  Similarly, ` F/_ x A ` is read ` x ` is not free in (class)
        ` A ` , see ~ df-nfc .</li>
 
-       <li>"$d x y $." should be read "Assume x and y are distinct
+       <li>"$d ` x y ` $." should be read "Assume ` x ` and ` y ` are distinct
        variables."</li>
 
-       <li>"$d x ` ph ` $." should be read "Assume x does not occur in phi $."
-       Sometimes a theorem is proved using
-       ` F/ x ph `  ( ~ df-nf ) in place of
-       "$d ` x ph ` $." when a more general result is desired;
+       <li>"$d ` ph x ` $." should be read "Assume ` x ` does not occur in
+       ` phi `."  Sometimes a theorem is proved using ` F/ x ph `  ( ~ df-nf )
+       in place of "$d ` ph x ` $." when a more general result is desired;
        ~ ax-5 can be used to derive the $d version.  For an example of
        how to get from the $d version back to the $e version, see the
        proof of ~ euf from ~ eu6 .</li>
 
-       <li>"$d x A $." should be read "Assume x is not a variable occurring in
-       class A."</li>
+       <li>"$d ` A x ` $." should be read "Assume ` x ` is not a variable
+       occurring in class ` A `."</li>
+
+       <li>"$d ` A x ` $.  $d ` ps x ` $.
+       $e |- ` ( x = A -> ( ph <-> ps ) ) ` $." is an idiom often used instead
+       of explicit substitution, meaning "Assume ` psi ` results from the
+       proper substitution of ` A ` for ` x ` in ` phi `."</li>
 
-       <li>"$d x A $.  $d x ps $.  $e |- ` ( x = A -> ( ph <-> ps ) ) ` $."
-       is an idiom
-       often used instead of explicit substitution, meaning "Assume psi results
-       from the proper substitution of A for x in phi."</li>
+       <li>Class and wff variables should appear at the beginning of distinct
+       variable conditions, and setvars should be in alphabetical order.
+       E.g., "$d ` Z x y ` $.",  "$d ` ps a x ` $.".  This convention should
+       be applied for new theorems (formerly, the class and wff variables
+       mostly appear at the end) and will be assured by a formatter in the
+       future.</li>
```

Comment:
It will be a pain to enforce as long as we don't have that formatter...

Reply:
Already existing theorems should not be touched, and for new theorems this should not be much efforts (metamath.exe does not create distinct variable conditions, and mmj2 and lamp already respect this convention.

Reply:
Was this discussed before, actually ?

Which alphabetical order to use ? For the Latin alphabet, it's commonly agreed that the first three letters are x, y, z. But already for the fourth letter, I think the most common is 't' but there are other conventions. Similarly for the Greek alphabet: phi, psi, chi are the first three, but after that there is no dominating convention.

If an order had to be adopted, I think it should be the order in which they are declared (with the $v statment), or better, the order of the $f-statements that type them (remembering these are local).

Reply:
> Was this discussed before, actually ?

Yes, see discussion in PR #3573, but also  https://github.com/expln/metamath-lamp/issues/172 and  https://groups.google.com/g/metamath/c/vRXuz7Qs5wY/m/eaxltrMsBwAJ 

The current convention concerns the order within a distinct variable condition: here, only the small Latin letters (for setvar variables) need to be sorted (alphabetically in the usual meaning, i.e. a b c ... x y z). This seems to be a convention which was silently followed already before (see, for example,  ~csbiebg: `$d a x A $.`).

Another issue is the ordering of the distinct variable conditions themselves. I think in most cases, the DV conditions (starting) with a class variable come first, ordered according to the class variables which are upper case Latin letters ( `$d A x y $.` ...` $d Z a b $.`), followed by  DV conditions (starting) with other class variables (special symbols like `.+` or `.(+)`), followed by DV conditions (starting) with wwf variables (greek letters, how ordered?) ending with DV conditions without class and wff variables (consisting of setvar variables only). This issue, however, was not discussed yet (at least as far as I know), and should be treated separately.